ui/src/main/java/edu/wpi/grip/ui/codeGeneration/Exporter.java

```diff
@@ -0,0 +1,138 @@
+   * @param lang the language to generate code for.
+   * @param dir the file to generate the main code to.
+   * @param testing if true enables features that allow for junit run tests for generated code.
+   */
+  public Exporter(ImmutableList<Step> steps, Language lang, File dir, boolean testing) {
+    this.steps = steps;
+    this.lang = lang;
+    this.dir = dir;
+    this.testing = testing;
+  }
+  
+  /**
+   * Constructor for an exporter for use when not testing.
+   * 
+   * @param steps an Immutable List of the steps in the pipeline to generate.
+   * @param lang the language to generate code for.
+   * @param dir the file to generate the main code to.
+   */
+  public Exporter(ImmutableList<Step> steps, Language lang, File dir) {
+    this(steps, lang, dir, false);
+  }
+  
+  @Override
+  public void run() {
+    TPipeline tPipeline = new TPipeline(steps);
+    TemplateMethods tempMeth = TemplateMethods.get(lang);
+    VelocityContext context = new VelocityContext();
+    context.put("pipeline", tPipeline);
+    context.put("tMeth", tempMeth);
+    context.put("fileName", dir.getName().substring(0, dir.getName().lastIndexOf(".")));
+    context.put("testing", testing);
+    String templateDir = "/edu/wpi/grip/ui/codegeneration/" + lang.filePath;
+    context.put("vmLoc", templateDir);
+    VelocityEngine ve = new VelocityEngine();
+    Properties props = new Properties();
+    props.put("velocimacro.library", templateDir + File.separator + "macros.vm");
+    props.put(RuntimeConstants.RESOURCE_LOADER, "classpath");
+    props.put("classpath.resource.loader.class", ClasspathResourceLoader.class.getName());
+    ve.init(props);
+    try {
+      generateCode(ve, templateDir, dir, context);
+      if (lang.equals(Language.CPP)) {
+        generateH(ve, templateDir, dir, context);
+      }
+    } catch (ResourceNotFoundException e) {
+      String error = e.getMessage();
+      String missingOperation = error.substring(error.lastIndexOf('/') + 1, error.lastIndexOf('.'));
+      logger.log(Level.SEVERE,
+          "The operation " + missingOperation + " is not supported for export to " + lang, e);
+      throw e; 
+    }
+  }
+
+  /**
+   * Creates a file and generates code in it using templates.
+   *
+   * @param ve The velocity engine used with the desired properties.
+   * @param templateDir The directory of the velocity templates that will be used.
+   * @param file The location to put the file.
+   * @param context The velocity context including the java files that will be used by the
+   *        templates.
+   */
+  private void generateCode(VelocityEngine ve, String templateDir, File file,
+      VelocityContext context) {
+    Template tm = ve.getTemplate(templateDir + File.separator + PIPELINE_TEMPLATE);
+    StringWriter sw = new StringWriter();
+    tm.merge(context, sw);
+    try (PrintWriter writer = new PrintWriter(file.getAbsolutePath(), "UTF-8")) {
```

Comment:
There should be a different constructor that takes an enumeration and doesn't throw the `UnsupportedEncoding` exception